scripts/i18n.js

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env node
+/**
+ * @fileoverview
+ * Updates the translations of the project, located in the
+ * directory locales/
+ */
+
+import fs from 'node:fs/promises';
+import fsSync from 'node:fs';
+import path from 'path';
+import { fileURLToPath } from 'node:url';
+import pugLex from 'pug-lexer';
+import PO from 'pofile';
+
+export const LANGUAGES = ['es', 'fr'];
+export const LANGUAGE_NAMES = {
+  en: 'English',
+  es: 'Español',
+  fr: 'Français',
+};
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const ROOT_DIR = path.resolve(__dirname, '..');
+const LOCALES_DIR = path.join(ROOT_DIR, 'locales');
+const INDEX_PATH = path.join(ROOT_DIR, 'public', 'index.pug');
+
+export const loadTranslations = () => {
+  const locales = {};
+
+  const pofiles = fsSync.readdirSync(LOCALES_DIR);
+  for (let fname of pofiles) {
+    if (!fname.endsWith('.po')) {
+      continue;
+    }
+
+    const locale = fname.slice(0, -3);
+    const pofile = path.join(LOCALES_DIR, fname);
+    const po = PO.parse(fsSync.readFileSync(pofile, 'utf8'));
+    const translations = {};
+    for (let item of po.items) {
+      translations[item.msgid] = item.msgstr[0];
+    }
+    locales[locale] = translations;
+
+    if (locales.en === undefined) {
+      const defaultTranslations = {};
+      for (let item of po.items) {
+        defaultTranslations[item.msgid] = item.msgid;
+      }
+      locales.en = defaultTranslations;
+    }
+  }
+
+  const i18n = (locale) => {
+    const translations = locales[locale];
+    const func = (msgid) => translations[msgid] || msgid;
+    func.translations = translations;
+    return func;
+  };
+
+  return i18n;
+};
+
+const extractTranslatableStringsFromIndex = (tokens) => {
+  const searchTokens = ['interpolated-code', 'attribute', 'code'];
+
+  const msgids = [];
+  let i = 0;
+  while (i < tokens.length) {
+    const token = tokens[i];
+    if (
+      searchTokens.includes(token.type) &&
+      typeof token.val === 'string' &&
+      token.val.includes('t_(')
+    ) {
+      let msgid = token.val.match(/(?!t_\(')([^']+)('\))/);
+      msgids.push(msgid[1]);
+    }
+    i++;
+  }
+  return msgids;
+};
+
+export const updateTranslations = async () => {
+  const index = await fs.readFile(INDEX_PATH, 'utf8');
+  const msgids = extractTranslatableStringsFromIndex(pugLex(index));
+  const currentIso = new Date().toISOString();
+
+  for (const lang of LANGUAGES) {
+    let po;
+    const poPath = path.join(ROOT_DIR, 'locales', `${lang}.po`);
+    if (!fsSync.existsSync(poPath)) {
```

Comment:
`fs.exist()` has been deprecated. Let's use `fs.stat()` or `fs.access()` instead.

But:

> Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile(), or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

So we can just use a `try-catch` instead of the `if (fs.exist(path)) else`.

Reference: https://nodejs.org/dist/latest/docs/api/fs.html#fsstatpath-options-callback